File: frontend/src/components/ProductivityIntelligence.js
import React, { useState, useEffect } from 'react';
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from './ui/card';
import { Button } from './ui/button';
import { Textarea } from './ui/textarea';
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from './ui/select';
import { Badge } from './ui/badge';
import { Alert, AlertDescription } from './ui/alert';
import { Tabs, TabsContent, TabsList, TabsTrigger } from './ui/tabs';
import { 
  Zap, 
  Clock, 
  Target, 
  TrendingUp, 
  AlertTriangle, 
  CheckCircle,
  ArrowRight,
  Calendar,
  BarChart3,
  Users,
  DollarSign,
  Lightbulb,
  Settings,
  Timer
} from 'lucide-react';

const API_BASE_URL = process.env.REACT_APP_BACKEND_URL || 'http://localhost:8001';

const ProductivityIntelligence = () => {
  // Component state
  const [prompt, setPrompt] = useState('');
  const [contextType, setContextType] = useState('comprehensive');
  const [workflowFocus, setWorkflowFocus] = useState('daily');
  const [loading, setLoading] = useState(false);
  const [analysis, setAnalysis] = useState(null);
  const [promptTemplates, setPromptTemplates] = useState({});
  const [activeCategory, setActiveCategory] = useState('daily_monitoring');
  const [dailyPriorities, setDailyPriorities] = useState(null);
  const [workflowSuggestions, setWorkflowSuggestions] = useState([]);
  const [error, setError] = useState(null);

  // Load data on component mount
  useEffect(() => {
    loadPromptTemplates();
    loadDailyPriorities();
    loadWorkflowSuggestions();
  }, []);

  const loadPromptTemplates = async () => {
    try {
      const response = await fetch(`${API_BASE_URL}/api/productivity/prompts`);
      const data = await response.json();
      
      if (data.status === 'success') {
        setPromptTemplates(data.prompt_categories);
      }
    } catch (error) {
      console.error('Error loading prompt templates:', error);
      // Set fallback templates
      setPromptTemplates({
        daily_monitoring: [
          "What needs my immediate attention today?",
          "Show me yesterday's performance against targets",
          "Which customers should I reach out to today?"
        ],
        workflow_optimization: [
          "Show me my most time-consuming data analysis tasks",
          "What routine decisions could CustomerMindIQ make automatically?",
          "Create my personalized daily dashboard"
        ]
      });
    }
  };

  const loadDailyPriorities = async () => {
    try {
      const response = await fetch(`${API_BASE_URL}/api/productivity/daily-priorities`);
      const data = await response.json();
      
      if (data.status === 'success') {
        setDailyPriorities(data.daily_priorities);
      }
    } catch (error) {
      console.error('Error loading daily priorities:', error);
    }
  };

  const loadWorkflowSuggestions = async () => {
    try {
      const response = await fetch(`${API_BASE_URL}/api/productivity/workflow-suggestions`);
      const data = await response.json();
      
      if (data.status === 'success') {
        setWorkflowSuggestions(data.workflow_suggestions);
      }
    } catch (error) {
      console.error('Error loading workflow suggestions:', error);
    }
  };

  const handleAnalyze = async () => {
    if (!prompt.trim()) {
      setError('Please enter a productivity analysis prompt');
      return;
    }

    setLoading(true);
    setError(null);
    setAnalysis(null);

    try {
      const response = await fetch(`${API_BASE_URL}/api/productivity/analyze`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          prompt: prompt.trim(),
          context_type: contextType,
          workflow_focus: workflowFocus
        })
      });

      const data = await response.json();

      if (response.ok) {
        setAnalysis(data);
      } else {
        throw new Error(data.detail || 'Analysis failed');
      }
    } catch (error) {
      console.error('Analysis error:', error);
      setError(error.message || 'Failed to generate analysis. Please try again.');
    } finally {
      setLoading(false);
    }
  };

  const handlePromptSelect = (selectedPrompt) => {
    setPrompt(selectedPrompt);
  };

  const formatAnalysis = (analysisText) => {
    if (!analysisText) return '';
    
    const sections = analysisText.split('\n\n');
    return sections.map((section, index) => {
      const lines = section.split('\n');
      const title = lines[0];
      const content = lines.slice(1).join('\n');
      
      if (title.includes('Executive Summary') || title.includes('Immediate Actions') || title.includes('Opportunities')) {
        return (
          <div key={index} className="mb-4">
            <h4 className="font-semibold text-blue-300 mb-2">{title}</h4>
            <div className="text-gray-300 whitespace-pre-wrap">{content}</div>
          </div>
        );
      }
      
      return (
        <div key={index} className="mb-3 text-gray-300 whitespace-pre-wrap">
          {section}
        </div>
      );
    }).filter(Boolean);
  };

  const getCategoryIcon = (category) => {
    const icons = {
      getting_started: Settings,
      daily_monitoring: Calendar,
      weekly_analysis: BarChart3,
      monthly_strategic: TrendingUp,
      immediate_attention: AlertTriangle,
      workflow_optimization: Zap
    };
    
    const IconComponent = icons[category] || Lightbulb;
    return <IconComponent className="h-4 w-4" />;
  };

  const getCategoryColor = (category) => {
    const colors = {
      getting_started: 'bg-green-500/20 text-green-300 border-green-500/30',
      daily_monitoring: 'bg-blue-500/20 text-blue-300 border-blue-500/30',
      weekly_analysis: 'bg-purple-500/20 text-purple-300 border-purple-500/30',
      monthly_strategic: 'bg-orange-500/20 text-orange-300 border-orange-500/30',
      immediate_attention: 'bg-red-500/20 text-red-300 border-red-500/30',
      workflow_optimization: 'bg-yellow-500/20 text-yellow-300 border-yellow-500/30'
    };
    
    return colors[category] || 'bg-gray-500/20 text-gray-300 border-gray-500/30';
  };

  const getPriorityColor = (priority) => {
    const colors = {
      Critical: 'bg-red-500/20 text-red-300 border-red-500/50',
      High: 'bg-orange-500/20 text-orange-300 border-orange-500/50',
      Medium: 'bg-yellow-500/20 text-yellow-300 border-yellow-500/50',
      Low: 'bg-green-500/20 text-green-300 border-green-500/50'
    };
    return colors[priority] || 'bg-gray-500/20 text-gray-300 border-gray-500/50';
  };

  return (
    <div className="space-y-6">
      {/* Header */}
      <Card className="bg-gray-900/50 border-gray-700">
        <CardHeader>
          <div className="flex items-center gap-3">
            <div className="p-2 bg-yellow-500/20 rounded-lg">
              <Zap className="h-6 w-6 text-yellow-400" />
            </div>
            <div>
              <CardTitle className="text-xl text-white">Daily Productivity Intelligence</CardTitle>
              <CardDescription className="text-gray-400">
                Start your day with AI-powered insights combining customer analytics and website data
              </CardDescription>
            </div>
          </div>
        </CardHeader>
      </Card>

      <div className="grid grid-cols-1 xl:grid-cols-3 gap-6">
        {/* Main Analysis Panel */}
        <div className="xl:col-span-2 space-y-6">
          {/* Quick Daily Priorities */}
          {dailyPriorities && (
            <Card className="bg-gradient-to-r from-red-600/10 to-orange-600/10 border-red-500/30">
              <CardHeader>
                <CardTitle className="text-lg text-white flex items-center gap-2">
                  <AlertTriangle className="h-5 w-5 text-red-400" />
                  Today's Priority Items
                </CardTitle>
              </CardHeader>
              <CardContent>
                <div className="space-y-3">
                  {dailyPriorities.urgent_attention?.critical_customers?.length > 0 && (
                    <div className="p-3 bg-red-500/10 rounded-lg border border-red-500/30">
                      <div className="flex items-center gap-2 mb-2">
                        <Users className="h-4 w-4 text-red-400" />
                        <span className="font-medium text-red-300">Critical Customers Need Attention</span>
                      </div>
                      <p className="text-gray-300 text-sm">
                        {dailyPriorities.urgent_attention.critical_customers.length} customers at high churn risk
                      </p>
                    </div>
                  )}
                  
                  {dailyPriorities.opportunities?.expansion_candidates?.length > 0 && (
                    <div className="p-3 bg-green-500/10 rounded-lg border border-green-500/30">
                      <div className="flex items-center gap-2 mb-2">
                        <DollarSign className="h-4 w-4 text-green-400" />
                        <span className="font-medium text-green-300">Expansion Opportunities</span>
                      </div>
                      <p className="text-gray-300 text-sm">
                        {dailyPriorities.opportunities.expansion_candidates.length} customers showing growth signals
                      </p>
                    </div>
                  )}
                </div>
              </CardContent>
            </Card>
          )}

          {/* Prompt Input */}
          <Card className="bg-gray-900/50 border-gray-700">
            <CardHeader>
              <CardTitle className="text-lg text-white flex items-center gap-2">
                <Target className="h-5 w-5 text-yellow-400" />
                Productivity Analysis Prompt
              </CardTitle>
            </CardHeader>
            <CardContent className="space-y-4">
              <div className="space-y-2">
                <label className="text-sm font-medium text-gray-300">Your Productivity Question</label>
                <Textarea
                  placeholder="Ask about daily priorities, workflow optimization, or strategic insights... (e.g., 'What should I focus on today to maximize impact?')"
                  value={prompt}
                  onChange={(e) => setPrompt(e.target.value)}
                  className="min-h-[100px] bg-gray-800 border-gray-600 text-white placeholder-gray-400"
                />
              </div>

              <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                <div>
                  <label className="text-sm font-medium text-gray-300">Data Context</label>
                  <Select value={contextType} onValueChange={setContextType}>
                    <SelectTrigger className="bg-gray-800 border-gray-600 text-white">
                      <SelectValue />
                    </SelectTrigger>
                    <SelectContent className="bg-gray-800 border-gray-600">
                      <SelectItem value="comprehensive">All Business Data</SelectItem>
                      <SelectItem value="priority_focus">Priority Items Only</SelectItem>
                      <SelectItem value="customer_data">Customer Data</SelectItem>
                      <SelectItem value="website_data">Website Data</SelectItem>
                    </SelectContent>
                  </Select>
                </div>

                <div>
                  <label className="text-sm font-medium text-gray-300">Workflow Focus</label>
                  <Select value={workflowFocus} onValueChange={setWorkflowFocus}>
                    <SelectTrigger className="bg-gray-800 border-gray-600 text-white">
                      <SelectValue />
                    </SelectTrigger>
                    <SelectContent className="bg-gray-800 border-gray-600">
                      <SelectItem value="daily">Daily Tasks</SelectItem>
                      <SelectItem value="weekly">Weekly Planning</SelectItem>
                      <SelectItem value="monthly">Monthly Strategy</SelectItem>
                      <SelectItem value="immediate">Immediate Actions</SelectItem>
                    </SelectContent>
                  </Select>
                </div>
              </div>

              <div className="flex justify-end">
                <Button 
                  onClick={handleAnalyze}
                  disabled={loading || !prompt.trim()}
                  className="bg-yellow-600 hover:bg-yellow-700"
                >
                  {loading ? (
                    <>
                      <Timer className="h-4 w-4 mr-2 animate-spin" />
                      Analyzing...
                    </>
                  ) : (
                    <>
                      <Zap className="h-4 w-4 mr-2" />
                      Get Insights
                    </>
                  )}
                </Button>
              </div>

              {error && (
                <Alert className="border-red-500/50 bg-red-500/10">
                  <AlertTriangle className="h-4 w-4 text-red-400" />
                  <AlertDescription className="text-red-300">
                    {error}
                  </AlertDescription>
                </Alert>
              )}
            </CardContent>
          </Card>

          {/* Analysis Results */}
          {analysis && (
            <Card className="bg-gray-900/50 border-gray-700">
              <CardHeader>
                <CardTitle className="text-lg text-white flex items-center gap-2">
                  <Lightbulb className="h-5 w-5 text-yellow-400" />
                  Productivity Analysis Results
                  <Badge className={`ml-auto ${getPriorityColor(analysis.priority_level)}`}>
                    {analysis.priority_level} Priority
                  </Badge>
                </CardTitle>
                <div className="flex items-center gap-4 text-sm text-gray-400">
                  <span>⏱️ {analysis.processing_time?.toFixed(2)}s</span>
                  <span>💾 {analysis.estimated_time_savings} potential savings</span>
                </div>
              </CardHeader>
              <CardContent>
                <Tabs defaultValue="analysis" className="w-full">
                  <TabsList className="grid w-full grid-cols-3 bg-gray-800">
                    <TabsTrigger value="analysis" className="text-gray-300">Analysis</TabsTrigger>
                    <TabsTrigger value="actions" className="text-gray-300">Action Items</TabsTrigger>
                    <TabsTrigger value="recommendations" className="text-gray-300">Recommendations</TabsTrigger>
                  </TabsList>
                  
                  <TabsContent value="analysis" className="mt-4">
                    <div className="space-y-4 max-h-96 overflow-y-auto">
                      {formatAnalysis(analysis.analysis)}
                    </div>
                  </TabsContent>
                  
                  <TabsContent value="actions" className="mt-4">
                    <div className="space-y-3">
                      {analysis.action_items?.length > 0 ? (
                        analysis.action_items.map((action, index) => (
                          <div key={index} className="flex items-start gap-3 p-3 bg-gray-800/50 rounded-lg">
                            <Target className="h-5 w-5 text-yellow-400 mt-0.5 flex-shrink-0" />
                            <div>
                              <p className="text-gray-300">{action}</p>
                              <Badge className="mt-2 bg-yellow-500/20 text-yellow-300 text-xs">
                                Action #{index + 1}
                              </Badge>
                            </div>
                          </div>
                        ))
                      ) : (
                        <p className="text-gray-400 italic">No specific action items extracted</p>
                      )}
                    </div>
                  </TabsContent>
                  
                  <TabsContent value="recommendations" className="mt-4">
                    <div className="space-y-3">
                      {analysis.recommendations?.length > 0 ? (
                        analysis.recommendations.map((rec, index) => (
                          <div key={index} className="flex items-start gap-3 p-3 bg-gray-800/50 rounded-lg">
                            <CheckCircle className="h-5 w-5 text-green-400 mt-0.5 flex-shrink-0" />
                            <p className="text-gray-300">{rec}</p>
                          </div>
                        ))
                      ) : (
                        <p className="text-gray-400 italic">No specific recommendations extracted</p>
                      )}
                    </div>
                  </TabsContent>
                </Tabs>
              </CardContent>
            </Card>
          )}
        </div>

        {/* Sidebar */}
        <div className="space-y-6">
          {/* Workflow Suggestions */}
          {workflowSuggestions.length > 0 && (
            <Card className="bg-gray-900/50 border-gray-700">
              <CardHeader>
                <CardTitle className="text-lg text-white flex items-center gap-2">
                  <Settings className="h-5 w-5 text-blue-400" />
                  Quick Wins
                </CardTitle>
              </CardHeader>
              <CardContent>
                <div className="space-y-3">
                  {workflowSuggestions.map((suggestion, index) => (
                    <div key={index} className="p-3 bg-gray-800/50 rounded-lg">
                      <div className="flex items-center justify-between mb-2">
                        <h4 className="font-medium text-white text-sm">{suggestion.category}</h4>
                        <Badge className={getPriorityColor(suggestion.priority)}>
                          {suggestion.priority}
                        </Badge>
                      </div>
                      <p className="text-gray-300 text-sm mb-2">{suggestion.suggestion}</p>
                      <div className="flex items-center gap-2 text-xs text-gray-400">
                        <Clock className="h-3 w-3" />
                        <span>Saves {suggestion.time_savings}</span>
                      </div>
                    </div>
                  ))}
                </div>
              </CardContent>
            </Card>
          )}

          {/* Prompt Templates */}
          <Card className="bg-gray-900/50 border-gray-700">
            <CardHeader>
              <CardTitle className="text-lg text-white flex items-center gap-2">
                <Lightbulb className="h-5 w-5 text-purple-400" />
                Productivity Prompts
              </CardTitle>
            </CardHeader>
            <CardContent>
              <div className="space-y-4">
                {/* Category Selection */}
                <div className="flex flex-wrap gap-2">
                  {Object.keys(promptTemplates).map((category) => (
                    <Badge
                      key={category}
                      variant={activeCategory === category ? "default" : "outline"}
                      className={`cursor-pointer transition-colors text-xs ${
                        activeCategory === category 
                          ? getCategoryColor(category)
                          : 'border-gray-600 text-gray-400 hover:text-gray-300'
                      }`}
                      onClick={() => setActiveCategory(category)}
                    >
                      {getCategoryIcon(category)}
                      <span className="ml-1 capitalize">
                        {category.replace('_', ' ')}
                      </span>
                    </Badge>
                  ))}
                </div>

                {/* Prompts List */}
                <div className="space-y-2 max-h-64 overflow-y-auto">
                  {promptTemplates[activeCategory]?.map((promptText, index) => (
                    <div
                      key={index}
                      className="p-3 bg-gray-800/50 rounded-lg cursor-pointer hover:bg-gray-800 transition-colors group"
                      onClick={() => handlePromptSelect(promptText)}
                    >
                      <p className="text-sm text-gray-300 group-hover:text-white">
                        {promptText}
                      </p>
                      <ArrowRight className="h-4 w-4 text-gray-500 group-hover:text-yellow-400 mt-1 opacity-0 group-hover:opacity-100 transition-opacity" />
                    </div>
                  ))}
                </div>
              </div>
            </CardContent>
          </Card>
        </div>
      </div>
    </div>
  );
};

export default ProductivityIntelligence;

File: backend/modules/productivity_intelligence.py
"""
Daily Productivity Intelligence Module
AI-powered productivity and workflow optimization system for Customer Mind IQ
"""

from fastapi import APIRouter, HTTPException, Depends
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime, timezone
import os
import uuid
from dotenv import load_dotenv
from motor.motor_asyncio import AsyncIOMotorClient
from emergentintegrations.llm.chat import LlmChat, UserMessage
import json

# Load environment variables
load_dotenv()

# MongoDB setup
MONGO_URL = os.getenv("MONGO_URL", "mongodb://localhost:27017")
DB_NAME = os.getenv("DB_NAME", "customer_mind_iq")
client = AsyncIOMotorClient(MONGO_URL)
db = client[DB_NAME]

# LLM setup
EMERGENT_LLM_KEY = os.getenv("EMERGENT_LLM_KEY")

router = APIRouter()

# Pydantic Models
class ProductivityInsightRequest(BaseModel):
    prompt: str = Field(..., description="Productivity analysis prompt")
    context_type: str = Field(default="comprehensive", description="Type of data context to include")
    workflow_focus: str = Field(default="daily", description="Focus area for productivity analysis")

class ProductivityInsightResponse(BaseModel):
    insight_id: str
    prompt: str
    analysis: str
    data_summary: Dict[str, Any]
    recommendations: List[str]
    action_items: List[str]
    priority_level: str
    estimated_time_savings: str
    created_at: datetime
    processing_time: float

# Productivity Prompt Templates
PRODUCTIVITY_PROMPTS = {
    "getting_started": [
        "What data sources should I connect first for the biggest impact?",
        "Show me the 5 most important dashboards for my business type",
        "What baseline metrics should I establish before making changes?"
    ],
    "daily_monitoring": [
        "What needs my immediate attention today?",
        "Show me yesterday's performance against targets",
        "Which customers should I reach out to today?",
        "What patterns emerged in today's data?",
        "Set tomorrow's priorities based on today's data"
    ],
    "weekly_analysis": [
        "What should I focus on this week to move the needle?",
        "Show me last week's wins and areas needing attention",
        "Summarize this week's business performance trends",
        "What experiments should I run next week?"
    ],
    "monthly_strategic": [
        "What are the top 3 opportunities to grow revenue next month?",
        "Which initiatives from last month delivered the best ROI?",
        "What warning signs suggest I should change strategy?"
    ],
    "immediate_attention": [
        "Alert me to any data anomalies that could indicate serious problems",
        "Show me customers with sudden behavior changes",
        "What revenue-critical issues need fixing this week?",
        "Which customers are showing expansion signals right now?",
        "What quick wins could I implement today?"
    ],
    "workflow_optimization": [
        "Show me my most time-consuming data analysis tasks",
        "What routine decisions could CustomerMindIQ make automatically?",
        "Create my personalized daily dashboard",
        "How much time is CustomerMindIQ saving me versus manual analysis?",
        "What insights am I getting now that I couldn't see before?"
    ]
}

# Enhanced Data Context Builders
async def get_comprehensive_context() -> Dict[str, Any]:
    """Get comprehensive business context combining customer and website data"""
    try:
        # Customer data
        total_customers = await db.customers.count_documents({})
        customer_segments = await db.customers.aggregate([
            {"$group": {"_id": "$segment", "count": {"$sum": 1}}},
            {"$sort": {"count": -1}}
        ]).to_list(length=10)
        
        # High-risk customers
        high_risk_customers = await db.customers.find(
            {"churn_risk": {"$in": ["High", "Critical"]}},
            {"customer_id": 1, "name": 1, "churn_risk": 1, "total_spent": 1}
        ).limit(5).to_list(length=5)
        
        # Revenue data
        revenue_pipeline = await db.customers.aggregate([
            {"$group": {"_id": None, "total_revenue": {"$sum": "$total_spent"}}},
        ]).to_list(length=1)
        total_revenue = revenue_pipeline[0]["total_revenue"] if revenue_pipeline else 0
        
        # Campaign data
        active_campaigns = await db.campaigns.count_documents({"status": "active"})
        campaign_performance = await db.campaigns.aggregate([
            {"$group": {"_id": "$status", "count": {"$sum": 1}}},
            {"$sort": {"count": -1}}
        ]).to_list(length=5)
        
        # Recent activity (last 24 hours)
        yesterday = datetime.now(timezone.utc).replace(hour=0, minute=0, second=0, microsecond=0)
        recent_activity = {
            "new_customers": await db.customers.count_documents({"created_at": {"$gte": yesterday}}),
            "support_tickets": await db.support_tickets.count_documents({"created_at": {"$gte": yesterday}}) if hasattr(db, 'support_tickets') else 0,
        }
        
        return {
            "customer_overview": {
                "total_customers": total_customers,
                "customer_segments": customer_segments,
                "high_risk_customers": high_risk_customers
            },
            "revenue_metrics": {
                "total_revenue": total_revenue,
                "avg_customer_value": total_revenue / total_customers if total_customers > 0 else 0
            },
            "marketing_status": {
                "active_campaigns": active_campaigns,
                "campaign_performance": campaign_performance
            },
            "daily_activity": recent_activity,
            "analysis_timestamp": datetime.now(timezone.utc).isoformat(),
            "context_type": "comprehensive"
        }
    except Exception as e:
        print(f"Error getting comprehensive context: {e}")
        return {
            "error": "Could not retrieve comprehensive data",
            "analysis_timestamp": datetime.now(timezone.utc).isoformat()
        }

async def get_priority_context() -> Dict[str, Any]:
    """Get high-priority items needing immediate attention"""
    try:
        # Critical customers
        critical_customers = await db.customers.find(
            {"churn_risk": "Critical"},
            {"customer_id": 1, "name": 1, "total_spent": 1, "last_activity": 1}
        ).limit(10).to_list(length=10)
        
        # Failed campaigns or low performing
        underperforming_campaigns = await db.campaigns.find(
            {"status": {"$in": ["failed", "low_performance"]}},
            {"name": 1, "status": 1, "target_segment": 1}
        ).limit(5).to_list(length=5)
        
        # Expansion opportunities (high-value, active customers)
        expansion_opportunities = await db.customers.find(
            {"$and": [
                {"total_spent": {"$gt": 1000}},
                {"churn_risk": {"$in": ["Low", "Medium"]}},
                {"engagement_score": {"$gt": 70}}
            ]},
            {"customer_id": 1, "name": 1, "total_spent": 1, "engagement_score": 1}
        ).limit(5).to_list(length=5)
        
        return {
            "urgent_attention": {
                "critical_customers": critical_customers,
                "underperforming_campaigns": underperforming_campaigns
            },
            "opportunities": {
                "expansion_candidates": expansion_opportunities
            },
            "analysis_timestamp": datetime.now(timezone.utc).isoformat(),
            "context_type": "priority_focus"
        }
    except Exception as e:
        print(f"Error getting priority context: {e}")
        return {
            "error": "Could not retrieve priority data",
            "analysis_timestamp": datetime.now(timezone.utc).isoformat()
        }

async def build_productivity_context(context_type: str) -> Dict[str, Any]:
    """Build comprehensive context for productivity analysis"""
    if context_type == "comprehensive":
        return await get_comprehensive_context()
    elif context_type == "priority_focus":
        return await get_priority_context()
    elif context_type == "customer_data":
        # Reuse from ai_business_insights
        from modules.ai_business_insights import get_customer_context
        return {"customer_data": await get_customer_context()}
    elif context_type == "website_data":
        # Basic website data - can be enhanced later
        return {
            "website_data": {"placeholder": "Website analytics integration"},
            "analysis_timestamp": datetime.now(timezone.utc).isoformat()
        }
    else:
        return await get_comprehensive_context()

async def generate_productivity_analysis(prompt: str, context_data: Dict[str, Any], workflow_focus: str) -> Dict[str, Any]:
    """Generate AI-powered productivity analysis"""
    if not EMERGENT_LLM_KEY:
        raise HTTPException(status_code=500, detail="AI service not configured")
    
    try:
        session_id = f"productivity_{uuid.uuid4().hex[:8]}"
        
        chat = LlmChat(
            api_key=EMERGENT_LLM_KEY,
            session_id=session_id,
            system_message=f"""You are an expert business productivity consultant and data analyst specializing in workflow optimization and daily business intelligence.

Your role is to help business owners and managers maximize their productivity by providing:
1. Actionable daily/weekly/monthly recommendations
2. Priority-focused task lists based on data
3. Time-saving automation suggestions
4. Early warning alerts for business issues
5. Opportunity identification for growth

Focus Area: {workflow_focus}

When analyzing data, provide:
- Immediate action items (what to do today)
- Priority levels (Critical/High/Medium/Low)
- Estimated time savings from recommendations
- Specific next steps with clear timelines
- Resource requirements for implementation

Structure your response with:
- Executive Summary (2-3 sentences)
- Immediate Actions Required (top 3-5 items)
- Opportunities to Pursue (growth/efficiency)
- Early Warning Alerts (potential issues)
- Automation Recommendations
- Estimated Time Impact

Keep responses practical, actionable, and focused on maximizing business productivity."""
        ).with_model("openai", "gpt-4o-mini")
        
        # Build context-aware prompt
        context_summary = f"""
Business Productivity Analysis Request:
Workflow Focus: {workflow_focus}

Current Business Context:
{json.dumps(context_data, indent=2, default=str)}

Productivity Question: {prompt}

Please provide a comprehensive productivity analysis with specific action items, priorities, and time-saving recommendations based on this data.
"""
        
        user_message = UserMessage(text=context_summary)
        analysis_response = await chat.send_message(user_message)
        
        # Extract action items and priority level
        action_items = []
        priority_level = "Medium"
        
        lines = analysis_response.split('\n')
        in_actions = False
        
        for line in lines:
            line = line.strip()
            if 'action' in line.lower() or 'immediate' in line.lower():
                in_actions = True
            elif in_actions and line.startswith(('•', '-', '*', '1.', '2.', '3.')):
                clean_action = line.lstrip('•-*123456789. ').strip()
                if clean_action and len(clean_action) > 10:
                    action_items.append(clean_action)
            
            # Detect priority level
            if 'critical' in line.lower() or 'urgent' in line.lower():
                priority_level = "Critical"
            elif 'high priority' in line.lower() or 'important' in line.lower():
                priority_level = "High"
        
        # Extract recommendations (reuse logic from ai_business_insights)
        recommendations = []
        for line in lines:
            line = line.strip()
            if any(keyword in line.lower() for keyword in ['should', 'recommend', 'consider', 'implement', 'focus on']):
                if len(line) > 20 and len(line) < 200:
                    recommendations.append(line)
        
        return {
            "analysis": analysis_response,
            "recommendations": recommendations[:5],
            "action_items": action_items[:7],
            "priority_level": priority_level,
            "estimated_time_savings": "2-5 hours/week",
            "context_used": context_data,
            "session_id": session_id
        }
        
    except Exception as e:
        print(f"Productivity analysis error: {e}")
        raise HTTPException(status_code=500, detail=f"Productivity analysis failed: {str(e)}")

# API Endpoints
@router.get("/prompts")
async def get_productivity_prompts():
    """Get categorized productivity prompt templates"""
    return {
        "status": "success",
        "prompt_categories": PRODUCTIVITY_PROMPTS,
        "total_prompts": sum(len(prompts) for prompts in PRODUCTIVITY_PROMPTS.values()),
        "focus_areas": list(PRODUCTIVITY_PROMPTS.keys())
    }

@router.post("/analyze", response_model=ProductivityInsightResponse)
async def analyze_productivity(request: ProductivityInsightRequest):
    """Generate AI-powered productivity insights"""
    start_time = datetime.now()
    
    try:
        # Build analysis context
        context_data = await build_productivity_context(request.context_type)
        
        # Generate AI analysis
        ai_result = await generate_productivity_analysis(
            request.prompt, 
            context_data, 
            request.workflow_focus
        )
        
        # Create insight record
        insight_id = f"productivity_{uuid.uuid4().hex[:12]}"
        processing_time = (datetime.now() - start_time).total_seconds()
        
        insight = ProductivityInsightResponse(
            insight_id=insight_id,
            prompt=request.prompt,
            analysis=ai_result["analysis"],
            data_summary=context_data,
            recommendations=ai_result["recommendations"],
            action_items=ai_result["action_items"],
            priority_level=ai_result["priority_level"],
            estimated_time_savings=ai_result["estimated_time_savings"],
            created_at=datetime.now(timezone.utc),
            processing_time=processing_time
        )
        
        return insight
        
    except Exception as e:
        print(f"Productivity analysis error: {e}")
        raise HTTPException(status_code=500, detail=f"Productivity analysis failed: {str(e)}")

@router.get("/daily-priorities")
async def get_daily_priorities():
    """Get daily priority items that need immediate attention"""
    try:
        priority_context = await get_priority_context()
        
        return {
            "status": "success",
            "daily_priorities": priority_context,
            "last_updated": datetime.now(timezone.utc).isoformat()
        }
        
    except Exception as e:
        print(f"Daily priorities error: {e}")
        return {
            "status": "error",
            "message": "Could not retrieve daily priorities",
            "last_updated": datetime.now(timezone.utc).isoformat()
        }

@router.get("/workflow-suggestions")
async def get_workflow_suggestions():
    """Get personalized workflow optimization suggestions"""
    try:
        # This would be enhanced with user behavior tracking
        suggestions = [
            {
                "category": "Morning Routine",
                "suggestion": "Start with 'What needs my immediate attention today?' prompt",
                "time_savings": "15 minutes",
                "priority": "High"
            },
            {
                "category": "Weekly Planning",
                "suggestion": "Use Monday planning prompts to focus on high-impact activities",
                "time_savings": "2 hours/week",
                "priority": "Medium"
            },
            {
                "category": "Data Analysis",
                "suggestion": "Automate routine reporting with custom dashboards",
                "time_savings": "5 hours/week",
                "priority": "High"
            }
        ]
        
        return {
            "status": "success",
            "workflow_suggestions": suggestions,
            "total_potential_savings": "7+ hours/week"
        }
        
    except Exception as e:
        print(f"Workflow suggestions error: {e}")
        return {
            "status": "error",
            "message": "Could not generate workflow suggestions"
        }

# Export router
__all__ = ["router"]